ztrouy/gondor-backend
# Cc/feature/34/fetch next appointment

### Purpose
To create an endpoint to allow Clinicians and Patients to get their next approved appointments

### Files Changed
- Created two serializers for Clinicians and Patients getting their next appointment in `serializers/appointments.py`
- Created an endpoint that allows them to get their next approved appointment in `views/appointments.py`


### To Test
Fetch, then run the following commands in the project root directory
```
pipenv shell
```
```
pipenv install
```
```
./seed_database.sh
```
Then confirm the following:
- When logged in as a Patient:
   - When you send a GET request to `http://localhost:8000/appointments/next` IF they have an approved appointment that is scheduled in the future, you will return that data and the assigned clinician
   - IF there are no future approved appointments you will return a message saying that there are no appointments found
- When logged in as a Clinician:
   - When you send a GET request to `http://localhost:8000/appointments/next` IF they have an approved appointment that is scheduled in the future, you will return that data and the assigned patient
   - IF there are no future approved appointments you will return a message saying that there are no appointments found
- When NOT logged in as a Patient or Clinician:
  - When you send a GET request to `http://localhost:8000/appointments/next` you will return a message saying that you are not authorized to view this data

Closes #34
